packages/react-router/src/router.ts

```diff
@@ -1773,8 +1786,7 @@ export class Router<
                     preload: !!preload,
                     context: parentContext,
                     location,
-                    navigate: (opts: any) =>
-                      this.navigate({ ...opts, from: match.pathname }),
+                    navigate: (opts: any) => this.navigate({ ...opts }),
```

Comment:
Could avoid the spread here yea?

`navigate: (opts: any) => this.navigate(opts as any)`